src/docfx/publish/PublishUrlMap.cs

```diff
@@ -132,16 +132,33 @@ private PublishUrlMapItem ResolveOutputPathConflicts(IGrouping<string, PublishUr
             }
 
             // redirection file is preferred than source file
-            var redirection = conflicts.FirstOrDefault(x => x.SourcePath.Origin == FileOrigin.Redirection);
-            if (redirection != null)
+            var redirections = conflicts.Where(x => x.SourcePath.Origin == FileOrigin.Redirection).OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer);
+            var nonRedirections = conflicts.Where(x => x.SourcePath.Origin != FileOrigin.Redirection)
+                .OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer)
+                .Select(x => x.SourcePath.Path.Value);
+            var redirection = redirections.FirstOrDefault();
+            var redirectionCount = redirections.Count();
+
+            if (redirectionCount == 0)
             {
-                _errors.Add(Errors.Redirection.RedirectedFileNotRemoved(redirection.SourcePath));
-                return redirection;
+                _errors.Add(Errors.UrlPath.OutputPathConflict(conflicts.First().OutputPath, conflicts.Select(x => x.SourcePath)));
+                return conflicts.OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer).Last();
             }
+            else if (redirectionCount == 1)
+            {
+                _errors.Add(Errors.Redirection.RedirectedFileNotRemoved(nonRedirections));
+                return redirection!;
+            }
+            else
+            {
+                if (conflicts.Count() > redirectionCount)
+                {
+                    _errors.Add(Errors.Redirection.RedirectedFileNotRemoved(nonRedirections));
+                }
 
-            // otherwise, prefer the one based on FilePath
-            _errors.Add(Errors.UrlPath.OutputPathConflict(conflicts.First().OutputPath, conflicts.Select(x => x.SourcePath)));
-            return conflicts.OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer).Last();
+                _errors.Add(Errors.UrlPath.OutputPathConflict(conflicts.First().OutputPath, conflicts.Select(x => x.SourcePath)));
+                return redirection!;
+            }
```

Comment:
```c#
            if (redirection != null && conflicts.Any(x => x.SourcePath.Origin != FileOrigin.Redirection))
            {
                // redirection file is preferred than source file
                _errors.Add(Errors.Redirection.RedirectedFileNotRemoved(redirection.SourcePath));
                return redirection;
            }

            return conflicts.OrderBy(x => x.SourcePath.Path, PathUtility.PathComparer).Last();
```

What if always report `output-path-conflict`, and report `redirected-file-not-removed` when conflicts contain both.
This make the logic easier to describe and maintain, and won't cause extra confusion to end user.

@live1206 @herohua @AngryBerryMS

Reply:
We need to inform the user that, a redirection has been added and the original file needs to be removed.
This removing action point needs to be stressed out, we had this requirement from user previously and confirmed with PM.